evaluation/gpqa/run_infer.py

```diff
@@ -165,27 +191,37 @@ def process_instance(
         # ======= Run the agent on the instance =======
         # Prepare instruction for the agent using suggested format in gpqa codebase
         instruction = f"""
-        What is the correct answer to this question:\n
-        {instance['question']}\n
+What is the correct answer to this question:\n
+{instance['question']}\n
 
-        Choices:\n
-        (A) {instance['choices'][0]}\n
-        (B) {instance['choices'][1]}\n
-        (C) {instance['choices'][2]}\n
-        (D) {instance['choices'][3]}\n
-        \n\n
+Choices:\n
+(A) {instance['choices'][0]}\n
+(B) {instance['choices'][1]}\n
+(C) {instance['choices'][2]}\n
+(D) {instance['choices'][3]}\n
+\n\n
 
-        MOST IMPORTANT: Format your response as follows:
-        <<FINAL_ANSWER||
-        <insert correct answer here, must be one of A, B, C, D> (Please dont use any additional characters. Just the letter of the correct answer (A/B/C/D).)
-        ||FINAL_ANSWER>>
+MOST IMPORTANT: Format your response as follows:
+<<FINAL_ANSWER||
+<insert correct answer here, must be one of A, B, C, D> (Please dont use any additional characters. Just the letter of the correct answer (A/B/C/D).)
+||FINAL_ANSWER>>
 
-        Additional Instructions:
-        - You should ONLY interact with the environment provided to you AND NEVER ASK FOR HUMAN HELP.
-        """
+Additional Instructions:
+- Do not try to solve the question in a single step. Break it down into smaller steps.
+- You should ONLY interact with the environment provided to you AND NEVER ASK FOR HUMAN HELP.
 
-        # NOTE: You can actually set slightly different instruction for different agents
-        instruction += AGENT_CLS_TO_INST_SUFFIX[agent.__class__.__name__]
+- SUPER IMPORTANT: When you have reported the answer to the user in the requested format, (and only once that is done) in the next turn, please run the following command: <execute_bash> exit </execute_bash>.
+- Again you are being told a million times to first report the answer in the requested format (see again below for reference) before exiting. DO NOT EXIT WITHOUT REPORTING THE ANSWER FIRST.
+    That is, when you have decided on the answer report in the following format:
+
+    <<FINAL_ANSWER||
+    <insert correct answer here, must be one of A, B, C, D> (Please dont use any additional characters. Just the letter of the correct answer (A/B/C/D).)
+    ||FINAL_ANSWER>>
+    <execute_bash> exit </execute_bash>
```

Comment:
This is duplicated with the codeact user response above, could we deduplicate?

Reply:
Because these were the code we used to run experiments in our paper, maybe we should leave them as is for reproducibility and improve in future iterations?

Reply:
We could, but because the strings are the same, wouldn't delivering be functionally equivalent?